### ISSUE TITLE
[Bug]: Get request with default headers fails

### ISSUE THREAD

**Issue by bitsal** [josueeduardo/rest-client#91]
Version: 1.6.0

I create a client like this:
```
RestClient.builder()
      .defaultHeader("Content-Type", "application/json")
      .build()
```
and when I do a get request like this:
```
client.get(<url>)
    .asObject(Map)
```
It fails in `ClientRequest:172` with `NullPointerException` due to request body is null for get request.
```
request.getBody().implicitContentType()
```
`request.getBody()` is null

**Expected result:**
Body should be checked on null value.

**Comment by bitsal** [comment]
**Workaround:**
```
RestClient client = RestClient.builder().build();
client.get(<url>).asObject(Map)
```
and POST requests
```
client.get(<url>).contentType(<CT>).asObject(Map)
```

**Comment by josueeduardo** [comment]
Fixed on 0beefe580e0000b6508dc0a507e3914c6036538c as part of 1.7.0
